src/wpublish/QueriesAndChunks.h

```diff
@@ -50,14 +53,19 @@ class QueryStatistics {
 public:
     using Ptr = std::shared_ptr<QueryStatistics>;
 
-    using TaskId = std::pair<QueryId, int>;  ///< Unique identifier for task is jobId, fragment number.
-    /// For maps of Tasks in a query. It would be nice to use an unordered_map, but a hash need to be defined.
-    using TaskMap = std::map<TaskId, wbase::Task::Ptr>;
+    /// The dictionary type for all tasks for this user query keyed by:
+    ///   [jobId][fragmentNum][templateId][subChunkId]
+    using JobIdKeyType = int;
+    using FragmentNumKeyType = int;
+    using TemplateIdKeyType = size_t;
+    using SubChunkIdKeyType = int;
+    using TaskMap =
+            std::map<JobIdKeyType,
+                     std::map<FragmentNumKeyType,
+                              std::map<TemplateIdKeyType, std::map<SubChunkIdKeyType, wbase::Task::Ptr>>>>;
```

Comment:
I'm not sure why you're going with a multilevel map instead of a std::pair or std::tuple key. The performance is going to be roughly the same for finding a unique element and accessing all the elements is a simple `for (auto&& taskElement : taskMap) { ... }`. This is probably better if stats for all subchunks in a single chunk are needed.
Also, I'd probaly put subchunk in front of templateId. Currently, I think it's a maximum of 2 templates per subchunk and the template queries tend to be very similar. Separating everything out by subchunkId is probably more useful than templateId.